pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -521,3 +584,55 @@ func createMessageWithPingAndWindowUpdate(t *testing.T, headerFields []hpack.Hea
 
 	return buf.Bytes()
 }
+
+// createMessageWithRST creates a message with the given number of header frames and RST frames,
+// and inserts RST frames with the given error code.
+func createMessageWithRST(t *testing.T, headerFields []hpack.HeaderField, headerFramesCount, rstFramesCount, rstFramesCountInMessage int, errCode http2.ErrCode, includeStatusCode ...bool) []byte {
+	var buf bytes.Buffer
+	framer := http2.NewFramer(&buf, nil)
+
+	headersFrame, err := usmhttp2.NewHeadersFrameMessage(headerFields)
+	require.NoError(t, err, "could not create headers frame")
+
+	for i := 0; i < headerFramesCount; i++ {
+		streamID := 2*i + 1
+
+		// Writing the header frames to the buffer using the Framer.
+		require.NoError(t, framer.WriteHeaders(http2.HeadersFrameParam{
+			StreamID:      uint32(streamID),
+			BlockFragment: headersFrame,
+			EndStream:     false,
+			EndHeaders:    true,
+		}), "could not write header frames")
+
+		if len(includeStatusCode) > 0 && includeStatusCode[0] {
+			statusOkHeaderField := []hpack.HeaderField{
+				{Name: ":status", Value: "200"},
+			}
+
+			dataFrame, err := usmhttp2.NewHeadersFrameMessage(statusOkHeaderField)
+			require.NoError(t, err, "could not create data frame")
+
+			// Writing the header frames to the buffer using the Framer.
+			require.NoError(t, framer.WriteHeaders(http2.HeadersFrameParam{
+				StreamID:      uint32(streamID),
+				BlockFragment: dataFrame,
+				EndStream:     false,
+				EndHeaders:    true,
+			}), "could not write header frames")
+		}
+
+		// Writing the data frame to the buffer using the Framer.
+		require.NoError(t, framer.WriteData(uint32(streamID), true, []byte{}), "could not write data frame")
+
+		if rstFramesCount > 0 {
+			for j := 0; j < rstFramesCountInMessage; j++ {
```

Comment:
why is the outer condition is `rseFramesCount > 0` but the for loop uses `rstFramesCountInMessage`?

Reply:
what do you think now?